### ISSUE TITLE
Hive build 1.2.0-beta7 on macOS 10.14.6 crashes when switching network protocol to macOS native. pcap is ok. May be my AVB environment is bad, but Hive should give a warning instead of crashing. Hive has to be reinstalled to run again.

### ISSUE THREAD

**Issue by AWHubGit** [christophe-calmejane/Hive#76]
Hive build 1.2.0-beta7 on macOS 10.14.6 crashes when switching network protocol to macOS native. pcap is ok. May be my AVB environment is bad, but Hive should give a warning instead of crashing. Hive has to be reinstalled to run again.
Attached avbdiagnose and Hive crash report.

[Hive_2019-10-16-150716_imacmoj.crash.log](https://github.com/christophe-calmejane/Hive/files/3735129/Hive_2019-10-16-150716_imacmoj.crash.log)

[info.dump.log](https://github.com/christophe-calmejane/Hive/files/3735135/info.dump.log)

_Originally posted by @AWHubGit in https://github.com/christophe-calmejane/Hive/issues/1#issuecomment-542761250_

**Comment by christophe-calmejane** [comment]
Thanks for the report, I'll take a look at this. But FYI, macOS native is currently not working properly due to errors in macOS AVBFramework itself. It's hopefully fixed in macOS Catalina (I yet have to test it). Meanwhile I suggest not using it (and I could display a warning in Hive about that).

**Comment by AWHubGit** [comment]
Please tell us if errors in macOS AVBFramework itself are fixed now in macOS Catalina.
I have a MOTU 1248 and MOTU LP32. Both act as AVB devices on switch Netgear GS724Tv4 without problems. I also want to use iMac 2017 as AVB device connected to the switch.
https://motu.com/techsupport/technotes/interfaceviaethernet
Do you know is this now possible with/in Catalina? Still any hope for AVB with Apple macOS?
Your comment is appreciated!

**Comment by christophe-calmejane** [comment]
Catalina testing is still on my todo list, it's something I have to deeply test (we are talking about the lowest code level of the library) in order to be sure all issues I reported to Apple are now fixed.

**Comment by christophe-calmejane** [comment]
Update:
I ran some tests on Catalina, most of the issues in the framework are fixed (macOS part) but there is one more thing that is missing from the SDK and I don't know if it will be available before next macOS release.
Nonetheless, I think I've fixed the remaining issues in my part of the code and it might be usable once again. I'll make a new Hive release soon if everything goes well.
